services/ctan/ctan.service.js

```diff
@@ -1,12 +1,13 @@
 import Joi from 'joi'
 import { renderLicenseBadge } from '../licenses.js'
 import { renderVersionBadge } from '../version.js'
-import { BaseJsonService } from '../index.js'
+import { BaseJsonService, InvalidResponse } from '../index.js'
 
 const schema = Joi.object({
   license: Joi.array().items(Joi.string()).single(),
   version: Joi.object({
-    number: Joi.string().required(),
+    number: Joi.string().allow('').required(),
+    date: Joi.string().allow(''),
```

Comment:
if we expect this to be present in the response but set to empty string to indicate "not set" then yes it should be `.required()`

Reply:
In https://github.com/muzimuzhi/shields/commit/80c1fe94c9106659ec010d6b5247edb103f8a377 I added `.required()` to `date` without adding `date` entry to tests. Surprisingly, all checks passed.  Is this the expected result?